modules/mpa/encode.c

```diff
@@ -0,0 +1,161 @@
+/**
+ * @file mpa/encode.c mpa Encode
+ *
+ * Copyright (C) 2016 Symonics GmbH
+ */
+
+#include <re.h>
+#include <baresip.h>
+#include <twolame.h>
+#include <string.h>
+#include <speex/speex_resampler.h>
+#include "mpa.h"
+
+#undef DEBUG 
+
+struct auenc_state {	
+	twolame_options *enc;
+	int channels;
+	SpeexResamplerState *resampler;
+};
+
+
+static void destructor(void *arg)
+{
+	struct auenc_state *aes = arg;
+
+	if (aes->enc)
+		twolame_close(&aes->enc);
+#ifdef DEBUG
+	debug("mpa: encoder destroyed\n");
+#endif
+}
+
+int mpa_encode_update(struct auenc_state **aesp, const struct aucodec *ac,
+		       struct auenc_param *param, const char *fmtp)
+{
+	struct auenc_state *aes;
+	struct mpa_param prm;
+	const struct aucodec *auc = aucodec_find("MPA", 90000, 1);
```

Comment:
this should not be necessary, the same object is available in the ``ac`` pointer.